src/python/pants/backend/python/tasks2/python_execution_task_base.py

```diff
@@ -130,6 +119,9 @@ def create_pex(self, pex_info=None):
 
         extra_pex_paths = [pex.path() for pex in pexes if pex]
 
+        if extra_pex_paths:
+          pex_info.pex_path = ':'.join(extra_pex_paths)
```

Comment:
how about using the new `PexInfo.merge_pex_path()` method here instead of the if + set?